### PR TITLE
Add ssms-backed choice-only RLSSM inference

### PR DESCRIPTION
**Pull request by krishnbera**
## Summary

Adds HSSM-side inference support for ssms-defined choice-only RL models, consumed through `RLSSMConfig.from_ssms_model()`.

## Changes

- Adds HSSM recognition and logp support for `inv_temp_softmax_2` and `inv_temp_softmax_3` decision processes.
- Wires ssms choice-only presets into `RLSSMConfig.from_ssms_model()` with shared multi-output computed `q0..qN` functions.
- Supports scalar response-only observed data in the RL likelihood builder while preserving RT+choice behavior.
- Adds RLSSM-local choice-only response validation for numeric, finite, integral labels inside declared `choices`.
- Adds choice-only active-lapse handling by converting scalar lapse probabilities to log probabilities before distribution construction.

## Dependency note

This PR depends on the ssms source-of-truth model definitions in lnccbrown/ssm-simulators#296. HSSM CI can still run against the current released `ssm-simulators`; tests that require the new real ssms choice-only presets skip when `2AB_RW_InvTempSoftmax` and `3AB_RW_InvTempSoftmax` are not available. The non-ssms unit coverage uses fake `ssms.rl` objects and should not depend on the unreleased ssms branch.

## Validation

- `PYTHONPATH=/Users/krishnbera/Documents/HSSMSpine/repos/.worktrees/ssm-simulators/choice-only-rl-inv-temp-softmax UV_PROJECT_ENVIRONMENT=/Users/krishnbera/Documents/HSSMSpine/_local/venvs/choice-only-rl-hssm uv run --frozen pytest tests/rl/test_choice_only_rl.py -v` passed with 13 tests.
- `UV_PROJECT_ENVIRONMENT=/Users/krishnbera/Documents/HSSMSpine/_local/venvs/choice-only-rl-hssm uv run --frozen pytest tests/rl/test_rlssm.py tests/rl/test_rl_likelihood_builder.py tests/rl/test_rlssm_config.py` passed with 119 tests.
- `UV_PROJECT_ENVIRONMENT=/Users/krishnbera/Documents/HSSMSpine/_local/venvs/choice-only-rl-hssm uv run --frozen pytest tests/test_hssm.py::test_is_choice_only_and_deadline tests/test_likelihoods_choice_only.py tests/distribution_utils/test_distribution_utils.py::test_lapse_distribution` passed with 14 tests.
- `UV_PROJECT_ENVIRONMENT=/Users/krishnbera/Documents/HSSMSpine/_local/venvs/choice-only-rl-hssm uv run --frozen ruff check src/hssm/rl/rlssm.py tests/rl/test_choice_only_rl.py tests/rl/test_rlssm.py` passed.

## Summary by CodeRabbit

* **New Features**
  * Added support for choice-only reinforcement learning models, including new built-in discrete-choice presets and a tutorial for using them.
  * Improved handling of computed parameters so multi-output calculations are shared more efficiently and support response-only inputs.

* **Bug Fixes**
  * Choice-only models now consistently apply the correct behavior when created from model-name strings.
  * Added stronger validation for responses and lapse values, with clearer errors and warnings for invalid data.

* **Documentation**
  * Added the new choice-only RLSSM tutorial to the documentation navigation.

### CODE REVIEW

File: src/hssm/rl/registry.py [hunk 1]
```diff
@@ -169,6 +172,51 @@ def copy(self) -> RLSSMRegistryEntry:
...
+            & (response_int < n_choices)
+        )
+
+        logits = beta[:, None] * q_values
+        safe_response = jnp.clip(response_int, 0, n_choices - 1)
+        chosen_logits = jnp.take_along_axis(
+            logits, safe_response[:, None], axis=1
+        ).squeeze(axis=1)
+        logp = chosen_logits - logsumexp(logits, axis=1)
+        return jnp.where(valid_response, logp, -jnp.inf)
```
[the remark this answers is not in the document]
Comment (krishnbera) [reply src/hssm/rl/registry.py:202]: Addressed in 2e543083. The choice-only inverse-temperature softmax base logp now preserves the incoming lan_matrix floating dtype instead of forcing float32, and tests/rl/test_choice_only_rl.py includes a float64 regression test for this path.

File: tests/rl/test_choice_only_rl.py [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+        ],
+    )
+    @pytest.mark.slow
+    def test_real_ssms_choice_only_rlssm_compiles_finite_active_lapse_logp(
+        self, model_name, n_choices
+    ):
+        """A real ssms choice-only preset can compile a finite active-lapse logp."""
+        self._require_choice_only_ssms_model(model_name)
+        hssm.set_floatX("float32", update_jax=True)
+
```
[the remark this answers is not in the document]
Comment (krishnbera) [reply tests/rl/test_choice_only_rl.py]: Addressed in 2e543083. The real ssms choice-only smoke test now saves the prior PyTensor floatX value before temporarily switching to float32 and restores it in a finally block with update_jax=True.

File: src/hssm/rl/likelihoods/builder.py [hunk 3]
```diff
@@ -551,11 +551,13 @@ def make_rl_logp_func(
...
 
-    # Pre-compute vmapped versions of all compute functions
+    # Pre-compute vmapped versions of unique compute functions. Several computed
+    # parameters may share the same function when one RL replay produces multiple
+    # decision parameters (for example q0, q1, q2).
     vmapped_compute_funcs = (
         {
-            param_name: jax.vmap(compute_func, in_axes=0)
-            for param_name, compute_func in ssm_logp_func.computed.items()
+            id(compute_func): jax.vmap(compute_func, in_axes=0)
```
Comment (cpaniaguam) [inline src/hssm/rl/likelihoods/builder.py]: I think the call to `id` can be dropped. Functions are hashable.

File: src/hssm/rl/registry.py [hunk 4]
```diff
@@ -169,6 +172,51 @@ def copy(self) -> RLSSMRegistryEntry:
...
+    _SSM_REGISTRY[f"inv_temp_softmax_{_n_choices}"] = {
+        "ssm_base_logp_func": _make_inv_temp_softmax_base_logp(_n_choices),
+        "list_params_ssm": [
+            "beta",
+            *(f"q{i}" for i in range(_n_choices)),
+        ],
+        "bounds_ssm": {"beta": (0.0, jnp.inf)},
+        "params_default_ssm": [1.0, *([0.0] * _n_choices)],
+        "response": ["response"],
+    }
```
Comment (cpaniaguam) [inline src/hssm/rl/registry.py]: Consider clarifying the intent of this block.
```suggestion
def _register_inv_temp_softmax_ssm(
    n_choices: int,
    beta_bounds: tuple[float, float] = (0.0, jnp.inf),
    beta_default: float = 1.0,
    response: list[str] | None = None,
) -> None:
    """Register a choice-only inverse-temperature softmax decision process."""
    response = response or ["response"]
    # Computed (learning-process-supplied) Q-value params, one per choice.
    q_params = [f"q{i}" for i in range(n_choices)]

    entry = {
        "ssm_base_logp_func": _make_inv_temp_softmax_base_logp(n_choices),
        "list_params_ssm": ["beta", *q_params],
        "bounds_ssm": {"beta": beta_bounds},
        "params_default_ssm": [beta_default, *([0.0] * n_choices)],
        "response": response,
    }
    _SSM_REGISTRY[f"inv_temp_softmax_{n_choices}"] = entry


_register_inv_temp_softmax_ssm(2)
_register_inv_temp_softmax_ssm(3)
```

**Review by cpaniaguam — APPROVED** [review]
This is great! I left two small observations.

File: src/hssm/rl/likelihoods/builder.py (hunk 3 above)
Comment (krishnbera) [reply src/hssm/rl/likelihoods/builder.py]: Addressed in a52c2137. The RL likelihood builder now uses the annotated compute function objects directly as the `vmapped_compute_funcs` and `computed_groups` keys, dropping the redundant `id(...)` calls while preserving the shared-compute grouping behavior.

File: src/hssm/rl/registry.py (hunk 4 above)
Comment (krishnbera) [reply src/hssm/rl/registry.py]: Addressed in a52c2137. The choice-only inverse-temperature softmax registry entries are now registered through `_register_inv_temp_softmax_ssm()`, which makes the intent explicit while keeping the `inv_temp_softmax_2` and `inv_temp_softmax_3` metadata unchanged.
